### PR TITLE
KAFKA-12808: Remove Deprecated Methods under StreamsMetrics

### PR DESCRIPTION
**Pull request by jlprat**
Removal of methods already deprecated since 2.5.
Adapt test to use the new alternative method.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java [hunk 1]
```diff
@@ -777,12 +777,12 @@ public void testLatencyMetrics() {
         final String entity = "entity";
         final String operation = "put";
 
-        final Sensor sensor1 = streamsMetrics.addLatencyAndThroughputSensor(scope, entity, operation, RecordingLevel.DEBUG);
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java:780]: You can also remove the `@SuppressWarnings("deprecation")` at the top of the method.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java [hunk 2]
```diff
@@ -797,11 +797,11 @@ public void testThroughputMetrics() {
         final String entity = "entity";
         final String operation = "put";
 
-        final Sensor sensor1 = streamsMetrics.addThroughputSensor(scope, entity, operation, RecordingLevel.DEBUG);
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java:800]: You can also remove the `@SuppressWarnings("deprecation")` at the top of the method.

**Review by cadonna — COMMENTED** [review]
Thank you for the PR, @jlprat !

The PR looks good to me. I had just two rather minor comments.

**Comment by jlprat** [comment]
You are right, I'll fix those later today

**Comment by jlprat** [comment]
Fixed pushed, `@SuppressWarnings("deprecation")` is now removed, Feel free to review @cadonna

**Comment by cadonna** [comment]
Restarted the build since the JDK 11 build failed with
```
Process 'Gradle Test Executor 123' finished with non-zero exit value 143
```

**Comment by jlprat** [comment]
Tests are now not passing, but they seem all unrelated to the change itself

**Comment by cadonna** [comment]
Test failures are unrelated.

**Review by cadonna — APPROVED** [review]
@jlprat Thank you for the PR!

**Comment by jlprat** [comment]
Thank you for the review @cadonna

**Pull request merged**